src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -13,6 +13,8 @@
   private double FAILURE_RATE = 0.0; //NOSONAR
 
   private int torpedoCount = 0;
+  //A random generator for the instance instead of creating a new at every fire
+  private Random generator = new Random();
```

Comment:
Én is erre jutottam, pedig a SonarLint valami SecureRandom-ot javasolt, de az kapásból nem volt jó neki :c